fileszip/filesziptest/filesziptest.go

```diff
@@ -0,0 +1,109 @@
+package filesziptest
+
+import (
+	"archive/tar"
+	"archive/zip"
+	"bytes"
+	_ "embed"
+	"io"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/xerrors"
+)
+
+//go:embed testdata/test.tar
+var testTarFileBytes []byte
+
+//go:embed testdata/test.zip
+var testZipFileBytes []byte
```

Comment:
review: I chose to `go:embed` these so that we don't end up with silly path errors.